### PR TITLE
Implementing Quickstart

### LINKED ISSUE #21: Quick Join functional
No functional for "Quick Join".
WARN|GameServer->OnUnhandledMessage:215| [bioaleks<5>-Administrator]|Unhandled message CQuickStartReqMessage.
Need to kill client process for close client.
Client see only "Connection" and nothing can do.

P.S. Before add this functional we can do "failed request" message.

### PR DESCRIPTION
**Pull request by Demon0no**
Closes #21.

### CODE REVIEW

File: src/Netsphere.Network/Message/Game/C2S.cs [hunk 1]
```diff
@@ -62,7 +62,7 @@ public class CLoginReqMessage : IGameMessage
     public class CQuickStartReqMessage : IGameMessage
     {
         [BlubMember(0)]
-        public byte Unk { get; set; }
+        public byte GameRule { get; set; }
```
Comment (wtfblub) [inline src/Netsphere.Network/Message/Game/C2S.cs]: Use the [enum](https://github.com/wtfblub/NetspherePirates/blob/dotnetty/src/Netsphere/Constants.cs#L154) and make sure to the data type is right for serialization. [Example](https://github.com/wtfblub/NetspherePirates/blob/7fc82bc5216eb34416cb8af4a2d804ac9de0d126/src/Netsphere.Network/Data/Game/RoomDto.cs#L19)

File: src/Game/Network/Services/ChannelService.cs [hunk 2]
```diff
@@ -124,8 +127,35 @@ await session.Player.ChatSession.SendAsync(new SChatMessageAckMessage(ChatType.C
         [MessageHandler(typeof(CQuickStartReqMessage))]
         public Task CQuickStartReq(GameSession session, CQuickStartReqMessage message)
         {
-            //ToDo - Logic
-            return session.SendAsync(new SServerResultInfoAckMessage(ServerResult.FailedToRequestTask));
+            if (session.Player.Channel == null)
+                return session.SendAsync(new SServerResultInfoAckMessage(ServerResult.FailedToRequestTask));
+
+            var rooms = session.Player.Channel.RoomManager.Select(r => r.Map<Room, RoomDto>()).ToArray();
```
Comment (wtfblub) [inline src/Game/Network/Services/ChannelService.cs]: This is not needed and just a waste of allocations. You can use the RoomManager directly.

**Comment by wtfblub** [comment]
Instead of taking a random room it should look for the most fitting room based on multiple conditions.

Conditions which should be in the first implementation:
- Team balance(Which room needs a new player the most e.g. 5vs2 needs a player more than a 4vs3)
- Rooms which are close to the end(10-15 seconds maybe?) should have a lower priority

Some more condition i would like to see sometime in the future but not needed for a first implementation: 
- Matching equipment(A player with guns should prefer gun rooms)
- Maybe slightly prefer rooms with people on the friendlist
- Level balance/Skill balance

**Comment by Demon0no** [comment]
Ehhh, I clearly am not used to working with git... 🤐
No idea how to remove that last commit and rename the one before.

**Pull request closed without merging**